backend/server.js

```diff
@@ -2,28 +2,105 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import crypto from 'crypto'
+import bcrypt from 'bcrypt'
+import listEndpoints from 'express-list-endpoints'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/authAPI'
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const User = mongoose.model('User', {
+  name: {
+    type: String,
+    minlength: 3,
+    unique: true,
+    required: true
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
+})
+
+const authenticateUser = async (req, res, next) => {
+  try {
+    const accessToken = req.header('Authorization')
+    const user = await User.findOne({ accessToken })
+
+    if (!user) {
+      throw 'Login error '
+    }
+    req.user = user
+    next()
+  } catch (err) {
+    const errorMessage = 'Try again'  
+    res.status(401).json({ errors: err.errors })
+  }
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
```

Comment:
You need to create your Express instance and cors rules before the first middleware. It should fix your problem 😄